module/modmanager/manager.go

```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
 	for key, val := range moduleEnvironment {
 		cmd.Env = append(cmd.Env, key+"="+val)
 	}
-	cmdOut, err := cmd.CombinedOutput()
 
-	resultLogger := logger.With("path", firstRunPath, "output", string(cmdOut))
+	stdOut, err := cmd.StdoutPipe()
 	if err != nil {
-		resultLogger.Errorw("command failed", "error", err)
 		return err
 	}
+	stdErr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	var combined []string
+	var mu sync.Mutex
 
-	resultLogger.Infow("command succeeded")
+	scanOut := bufio.NewScanner(stdOut)
+	go func() {
+		for scanOut.Scan() {
+			out := scanOut.Text()
+			logger.Infow("first run stdio", "output", out)
```

Comment:
double check that this output gets sent to app.viam, I think we may have to use Infof so it's not lost

Reply:
i'm going to address the app.viam bug in a separate PR - unfortunately I think it might have something to do with my `AsZap` conversation, that I needed to do in order to call the `With` method.

Reply:
I think this is a blocker for merge, we should just fix it now and be done with it

Reply:
fixed https://github.com/viamrobotics/rdk/pull/4475/commits/bfc1c466223bb6848cbee61b9122811685d21cbd

it was the `AsZap` conversion as I suspected. weirdly I got an error when I didn't do that conversion initially - maybe it's fine now because we grab a sublogger first?

Reply:
ugh nvm, I still get a test failure when I don't convert to `AsZap`. super weird because it seems to work fine in non-test environments

sorry @dgottlieb @cheukt, I asked for a re-review prematurely. need to resolve this matter first

Reply:
the failure is a bit hard to spot because it involves a panic, but essentially it panics because I try to call `.WithFields` which seems to hit [this logic](https://github.com/viamrobotics/rdk/blob/1930eb10ab992374f3d8206a16dc93f62a873b77/logging/logger.go#L114-L117) in tests but if fine otherwise.

Reply:
fixed by [implemented the missing method](https://github.com/viamrobotics/rdk/pull/4475/commits/0808575a2181cfa334f3754b0a06798776a3aeb3) - confirmed that logs still go to app.viam

Reply:
I'm a little lost, but in the end we wound up removing the `AsZap`. That seems right to me. Was the only motivation for using `AsZap` in the original because we missed the existance of [WithFields](https://github.com/viamrobotics/rdk/blob/a6e455af13cc1a85e9421e65ce64be94d7aec16b/logging/logger.go#L23)?

The [missing method](https://github.com/viamrobotics/rdk/pull/4475/commits/0808575a2181cfa334f3754b0a06798776a3aeb3) only applies to `zLogger`s and I see nothing here that is [using](https://github.com/viamrobotics/rdk/blob/a6e455af13cc1a85e9421e65ce64be94d7aec16b/logging/logger.go#L84) `FromZapCompatible`

Reply:
So if I remove the change you reference [in this comment](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1811694898), I see a panic in our tests that points the line that calls `WithFields`.